### DISCUSSION TITLE
Running with volatile storage

### DISCUSSION THREAD

**Discussion by juan-castrillon** [dani-garcia/vaultwarden#3064] [Q&A] [answered]
Hi!

I am currently just learning about this tool and evaluating if it can be used for my needs. 

In that context, i wanted to deploy it as a **stateless** service (Using Hashicorp's [Nomad](https://developer.hashicorp.com/nomad/docs)) in a cluster that I have running. Next to it there is a static postgres instance. 

My idea was to deploy Vaultwarden with a postgres backend, but I would have to avoid mounting the `/data/` directory as I see in all the articles in the wiki. 

My question is, what will fail if the `/data` directory is not persistant, but the database is? 

Thanks in advance :)

**Comment by BlackDex** [comment, marked as answer]
The `/data` directory contains the priv/pub keys to generate JWT tokens. These tokens are used during the login and for tokens during the invite. If these files are removed/replaced (because the get generated again if they arn't there) then those previous tokens arn't valid anymore. They are not used for encryption of the vault or anything.

Also, favicon's are stored there as cache. You may want to use a different icon service in this case or maybe even disable it fully.
Same btw for attachments, those are stored on the data directory, both for vault and send.

**Reply by juan-castrillon** [reply]
I see.  I was just curious as I've testing it out without this mount and things seemed to be working fine. Thanks for the good explanation
